### DISCUSSION TITLE
Habit Stacking - Multiple habits under a main habit

### DISCUSSION THREAD

**Discussion by LexD90** [iSoron/uhabits#878] [Feature Requests] [answered]
I've been reading Atomic Habits recently and thought that habit stacking within the app would be amazing. Instead of cluttering/overwhelming one with multiple habits, it would be good to see a Main Habit e.g. Morning Coffee Routine have multiple sub-habits e.g.

- [ ] Morning Coffee Routine
     - [ ] Wake at 5am
     - [ ] Boil Kettle
     - [ ] Clear Kitchen Counter
     - [ ] Empty Dishwasher
     - [ ] Brew Coffee

You should be able to complete the full Morning Routine by just checking off the main habit for that day, or be able to do it individually (selecting main overrides current selections in sub-habits)

I would have multiple "Main" Habits as I try to do more Habit Stacking

Hope I'm not repeating a previous submission there's a few to get through. Thanks for the app!

**Comment by lukaszlew** [comment]
Indeed, for people that split their habits into small pieces (eg, each gym exercise separately, each part of house cleaning, etc) stacking is probably THE main habit cue. Such lists can be currently managed with colors, but this is indeed suboptimal.

The idea needs some UI hashing out, but overall IMO it is excellent.

**Reply by hiqua** [reply]
I feel this is mostly https://github.com/iSoron/uhabits/discussions/495

**Comment by iSoron** [comment, marked as answer]
Locking as duplicate of #495
